rust/src/router/direct_transport.rs

```diff
@@ -533,19 +564,15 @@ impl DirectTransport {
         let subscription_handler = {
             let handlers = Arc::clone(&handlers);
 
-            channel.subscribe_to_notifications(id.into(), move |notification| {
-                match serde_json::from_slice::<Notification>(notification) {
+            channel.subscribe_to_fbs_notifications(id.into(), move |notification| {
+                match Notification::from_fbs(notification) {
```

Comment:
@nazar-pc, seems this is the way to go. Definitely cleaner as each class will just care about it's own notifications.

Once confirmed this is the way to go, I'll implement in the rest of transports.

Reply:
This side looks fine, btu outer side seems to still have all events in a flat data structure, which is what I wanted to avoid. Basically when/if we add another notification in flatbuffers and C++ I want Rust to stop compiling and tell you to handle another enum variant. Currently this is not the case due to `_ =>` above.

Reply:
That is my biggest concern here